### DISCUSSION TITLE
Problem with mapped type Integer in Number column

### DISCUSSION THREAD

**Discussion by ederfmatos-exacta** [hibernate/hibernate-orm#7518] [Design Proposals]
I have a column in an Oracle database, version 19, defined as number(10). This allows for values up to a certain limit, 2,147,483,647. In practice, when a query is executed where the column value exceeds the maximum limit for an integer type, an error occurs. Within the code, I found the segment that designates the integer type and an accompanying comment stating that using the integer type is safe. However, this has not been my experience, as the integer type in Java has a maximum value of 2,147,483,647. Given that there's a significant chance of exceeding this limit, it seems that the number(10) definition might not be appropriate and that larger values might be anticipated. Indeed, in cases where a number(10) exceeds the integer range, errors can arise. In my situation, casting the column to a larger numeric type resolved the issue. It would be beneficial for this to be the default behavior to prevent such errors, especially in integration with Hibernate where type mappings are crucial.

https://github.com/hibernate/hibernate-orm/blob/408777469148384e78080ac0e5309d73f1bde846/hibernate-core/src/main/java/org/hibernate/dialect/OracleDialect.java#L820

**Comment by gavinking** [comment]
This has already been discussed and we will not change it. Hibernate needs to be able to correctly reverse-engineer types from the DDL it generates.

If you are using `NUMBER(10,0)` to store integers larger than 2,147,483,647, you're already doing something dangerous. You should _always_ use numeric types with plenty of headroom. If you anticipate numbers with ten digits, then use `NUMBER(11)` _at least_.
